src/MSBuild/XMake.cs

```diff
@@ -2545,6 +2545,39 @@ private static bool WarningsAsErrorsSwitchIsEmpty(CommandLineSwitches commandLin
             return indexOfColon < 0 || indexOfColon == val.Length - 1;
         }
 
+        internal static ProjectIsolationMode ProcessIsolateProjectsSwitch(string[] parameters)
+        {
+
+            // Before /isolate had parameters, it was treated as a boolean switch.
+            // Preserve that in case anyone is using /isolate:{false|true}
+            if (parameters.Length == 1 && bool.TryParse(parameters[0], out bool boolValue))
+            {
+                return boolValue ? ProjectIsolationMode.True : ProjectIsolationMode.False;
+            }
+
+            ProjectIsolationMode isolateProjects = ProjectIsolationMode.True;
+            foreach (string parameter in parameters)
```

Comment:
ProcessBooleanSwitch only looks at the last thing in the array. Why are you looping through it here? I don't know how to construct this in practice, but I'm imagining if this were "[foo, true]", then you'd be ok before but fail now.

Reply:
Yeah that's the idea. It's to ensure we error if `/isolate:foo /isolate:true` or `/isolate:true /isolate:foo` are specified in the command line. Without my change, it would not fail on `/isolate:foo /isolate:true`.